docs/api.yml

```diff
@@ -18,26 +18,41 @@ paths:
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/metadata'
+                $ref: "#/components/schemas/metadata"
               example:
                 {
-                  'documentId': 'kjsak2',
-                  'firstName': 'The name',
-                  'dob': '1999-01-01',
+                  "documentId": "kjsak2",
+                  "firstName": "The name",
+                  "dob": "1999-01-01",
                 }
   /{documentId}:
     get:
       operationId: get-metadata
       summary: Gets a document from S3 using the document id
       parameters:
-        - $ref: '#/components/parameters/documentId'
+        - $ref: "#/components/parameters/documentId"
       responses:
         200:
           description: OK
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/metadata'
+                $ref: "#/components/schemas/metadata"
+  /{documentId}/contents:
+    get:
+      operationId: get-document-contents
+      summary: >
+        Gets the contents of a document from S3 using the document id
+      parameters:
+        - $ref: "#/components/parameters/documentId"
+      responses:
+        302:
+          description: Found
+          headers:
+            Location:
+              schema:
+                type: string
+              description: The location of the document contents
```

Comment:
This will 302 to the S3 signed URL to fetch the document, this hides the fact it is stored in S3 from the client so long as they support following redirects.

Reply:
Out of interest, when would this not work? Really old browser?

Reply:
I think browsers should be safe, some HTTP clients don't follow redirects automatically - so if someone was doing an API call to fetch the document it could be confusing (also maybe `wget` and `curl` don't by default, I can never remember). But usually just a one line thing to tell it to follow the redirects and you're good.